cmd/main.go

```diff
@@ -250,16 +250,20 @@ func main() {
 		lead = &leader.DummyElector{}
 		autoMigrationLeader = lead
 		// in on-prem mode, setup file system s3 driver and use localjob implementation
-		objectHandler = s3wrapper.NewFSClient("/data", log)
+		objectHandler = s3wrapper.NewFSClient("/data", log, versionHandler)
 		createS3Bucket(objectHandler)
-		generator = job.NewLocalJob(log.WithField("pkg", "local-job-wrapper"), Options.JobConfig)
+		generator = job.NewLocalJob(log.WithField("pkg", "local-job-wrapper"), Options.JobConfig, versionHandler)
 		if Options.DeployTarget == deployment_type_ocp {
 			ocpClient, err = k8sclient.NewK8SClient("", log)
 			failOnError(err, "Failed to create client for OCP")
 		}
 
-		err = objectHandler.UploadBootFiles(context.Background())
-		failOnError(err, "Failed uploading boot files")
+		for version := range openshiftVersionsMap {
+			go func(version string) {
+				failOnError(objectHandler.UploadBootFiles(context.Background(), version),
+					"Failed uploading boot files for OCP version %s", version)
+			}(version)
+		}
```

Comment:
Need to make sure to defer apiEnabler.Enable() until the upload is complete.
Also, I would move this code down to where the other DeployTargets upload boot files.